app/components/Login/Login.jsx

```diff
@@ -47,6 +47,13 @@ export default class Login extends React.Component {
             this.setState({
                 openSettings: true
             });
+        } else {
+            // If a token was supplied in the window.suppliedAuthToken, simulate a login
+            if ( window.suppliedAuthToken ) {
+                this.setState({loginMethodType: 'TOKEN', authToken: window.suppliedAuthToken}, function () {
+                    this.validateToken({keyCode: 13});
```

Comment:
Nice work. 

In this case, if the token supplied by the header, which we assume as the SSO entity, fails, it would generate the generic "invalid" login. Can we be more explicit that if this step fails, it fails because of the automated login? This will help users understand to look downstream for the resolution.